### ISSUE TITLE
keras.engine.topology

### ISSUE THREAD

**Issue by starboyvarun** [keras-team/keras#15238]
![2](https://user-images.githubusercontent.com/26819449/130608888-a88a346c-242e-4f65-b524-133460a8cda8.JPG)

I have installed all packages separately as mentioned. Still got this Error.

![2 1](https://user-images.githubusercontent.com/26819449/130608879-570d7983-071a-4e04-bb39-9151d28f2cd8.JPG)
Even did this as mentioned in stack overflow:
from tensorflow.keras.layers import Layer, InputSpec
I tried every possible thing to solve it which is on the internet.
Could you please help?
Thankyou

**Comment by tarun-bisht** [comment]
What TensorFlow version are you using?

**Comment by tarun-bisht** [comment]
```python
import tensorflow as tf
print(tf.__version__)
print(tf.keras.__version__)
```

**Comment by starboyvarun** [comment]
Actually, the application I was trying to run wanted to use  tensorflow==2.2 and keras==2.4.3
but when I tried this command.
![4 1](https://user-images.githubusercontent.com/26819449/130611950-dc7bd007-2448-4e7c-aeb6-ecb521221df6.JPG)

My version is still this :
![4](https://user-images.githubusercontent.com/26819449/130611649-24111077-5d83-47ae-9f87-48d508856593.JPG)

**Comment by tarun-bisht** [comment]
If you are working locally then create a new environment 
```bash
conda create --name env_name_here
conda install tensorflow==2.2 and keras==2.4.3
```
also I noticed that you said you want version tensorflow==2.2 and keras==2.4.3 but you are installing tensorflow 1.4 and keras 2.2

**Comment by tarun-bisht** [comment]
I am running tensorflow 2.4 and keras 2.4 for me 
```python
from tensorflow.keras.layers import Layer, InputSpec
```
works fine.

**Comment by tarun-bisht** [comment]
also error says that these versions are incompatible if you want tensorflow greater than 2.0 then you do not need to install keras, its part of tensorflow.

**Comment by starboyvarun** [comment]
![5](https://user-images.githubusercontent.com/26819449/130620259-4e0b6266-fea4-4fe9-8fe0-91df7ffa9b30.JPG)

I already used 2.4 and 2.4 .3 . this Error comes then.

![5 1](https://user-images.githubusercontent.com/26819449/130620281-dcf4886c-fb98-4e0c-80fa-fc11d2d4483a.JPG)
And see this is not working for me.
Thank You.

**Comment by tarun-bisht** [comment]
Please confirm your version again, In above screenshot your version says 2.6 but you told you used 2.4. Also try
```python
from keras.engine.input_spec import InputSpec
from keras.engine.base_layer import Layer
```
if that works.

**Comment by tarun-bisht** [comment]
I also checked Janggu dependencies they want you to have tensorflow 2.2 and keras 2.4.
First install these specific versions. Also tensorflow 2.2 requires python 3.8.
```python
conda install tensorflow==2.2 keras==2.4.3
```
I also tried this in colab with above version in google. Colab and its working for me
![Screenshot_20210824-201347~2](https://user-images.githubusercontent.com/47741102/130638768-90fff8dc-d7fd-45b0-bee9-e578e2c15fa5.png)
![Screenshot_20210824-201356~2](https://user-images.githubusercontent.com/47741102/130638798-44595fc3-7a4e-4317-b66e-ccfb897638fd.png)
![Screenshot_20210824-201407~2](https://user-images.githubusercontent.com/47741102/130638831-30054dae-fd26-4a7c-a80e-51346b851f47.png)
